### ISSUE TITLE
recursive in Gitdata - getTree

### ISSUE THREAD

**Issue by taphuocanh** [octokit/octokit.js#769]
recursive in Gitdata - getTree is not work, ?

**Comment by gr2m** [comment]
Hi @phuocanh421994, could you please share a code example?

**Comment by taphuocanh** [comment]
```js
const octokit = require('@octokit/rest')();
router.get('/all', (req, res, next) => {
  var owner = 'phuocanh421994';
  var repo = 'ndoejs.taphuocanh.com';
  var sha = '15b9207397a5611016b90a41f2154d0cc38e585b';
  var recursive = 1;
  var all ={};
  octokit.gitdata.getTree( { owner, repo, sha, recursive } ).then(result => {
    res.send(result);
  });
});
```

I want to receive a reponse that is

```json
{
  "sha": "15b9207397a5611016b90a41f2154d0cc38e585b",
  "url": "https://api.github.com/repos/phuocanh421994/ndoejs.taphuocanh.com/git/trees/15b9207397a5611016b90a41f2154d0cc38e585b",
  "tree": [
    {
      "path": "Khai-thac-toi-da-suc-manh-cua-Nodejs",
      "mode": "040000",
      "type": "tree",
      "sha": "957129480d6e7b569af133e6aae4a993ba014a12",
      "url": "https://api.github.com/repos/phuocanh421994/ndoejs.taphuocanh.com/git/trees/957129480d6e7b569af133e6aae4a993ba014a12"
    },
    {
      "path": "Nodejs-cho-nguoi-moi-bat-dau",
      "mode": "040000",
      "type": "tree",
      "sha": "af33f33170973627fcc0a15b4e455af6c2ec3026",
      "url": "https://api.github.com/repos/phuocanh421994/ndoejs.taphuocanh.com/git/trees/af33f33170973627fcc0a15b4e455af6c2ec3026"
    },
    {
      "path": "README.md",
      "mode": "100644",
      "type": "blob",
      "sha": "e69de29bb2d1d6434b8b29ae775ad8c2e48c5391",
      "size": 0,
      "url": "https://api.github.com/repos/phuocanh421994/ndoejs.taphuocanh.com/git/blobs/e69de29bb2d1d6434b8b29ae775ad8c2e48c5391"
    },
    {
      "path": "Sau-hon-ve-Nodejs",
      "mode": "040000",
      "type": "tree",
      "sha": "012d586cf0a04a3cfdf0341962a6c636a78051be",
      "url": "https://api.github.com/repos/phuocanh421994/ndoejs.taphuocanh.com/git/trees/012d586cf0a04a3cfdf0341962a6c636a78051be"
    },
    {
      "path": "Xay-dung-ung-dung-web-voi-Nodejs",
      "mode": "040000",
      "type": "tree",
      "sha": "4da8c2e419969e0bc22b5a028f0517f6a90041e8",
      "url": "https://api.github.com/repos/phuocanh421994/ndoejs.taphuocanh.com/git/trees/4da8c2e419969e0bc22b5a028f0517f6a90041e8"
    }
  ],
  "truncated": false
}
```
Thank

**Comment by gr2m** [comment]
Here is a test case, works as expected as far as I can tell?

```
const octokit = require('@octokit/rest')()

const owner = 'phuocanh421994'
const repo = 'ndoejs.taphuocanh.com'
const sha = '15b9207397a5611016b90a41f2154d0cc38e585b'
const recursive = 1
const all ={}
const {data: {tree}} = await octokit.gitdata.getTree( { owner, repo, sha, recursive } )

console.log(tree)
```

See https://runkit.com/gr2m/5a95993f169f37001262f57f

I think the problem is that you pass the entire result instead of just the data to `res.send`. 

Try this

```diff
  octokit.gitdata.getTree( { owner, repo, sha, recursive } ).then(result => {
-   res.send(result);
+   res.send(result.data);
  });
```

**Comment by taphuocanh** [comment]
You do not seem to understand my problem here. I would like to receive less return data with non recursive attributes. But it is not working when I set up: `recursive = false`. 
While the data now returns 35 records, I would like to get only 5 records.

**Comment by gr2m** [comment]
Thanks @phuocanh421994, I got it now. 

It turns out that this is most likely a Bug with GitHub’s API, not the `@octokit/rest` client. See https://api.github.com/repos/phuocanh421994/ndoejs.taphuocanh.com/git/trees/15b9207397a5611016b90a41f2154d0cc38e585b?recursive=0 for example. It should return the same as https://api.github.com/repos/phuocanh421994/ndoejs.taphuocanh.com/git/trees/15b9207397a5611016b90a41f2154d0cc38e585b because `recursive` is set to a falsy value, but it doesn’t work as expected.

I raised the issue with the GitHub team.

To workaround this problem, only pass the `recursive` argument if you want the recursive tree, otherwise leave it out entirely
